docs/HOL_Fault-injection-for-Azure-Container-Services(ACS)/README.md

```diff
@@ -0,0 +1,168 @@
+
+
+## Pre-requisites:
+
+- HOL - Deploying Parts Unlimited MRP on DC/OS
+
+- Access to VSTS to run a load test
+
+
+
+## Tasks Overview:
+**1. Set up load test** This will walk your through creating and configuring a load test for the next task using VSTS.
+
+**2. Inject faults into ACS environment using UI** In this task you will simulate faults of different parts of the system by killing its instances in Marathon service installed on DC/OS.
+
+**3. Inject faults into ACS environment using API** In this task you will learn how to achieve exactly the same outcome as in the second task but using Marathon's API.
+
+**4. Analyze results** In this task you will collect all of the information related to the load test and will draw a conclusion about the stability of the environment.
+
+
+
+## HOL:
+
+### Task 1: Set up load test
+
+**Step 1.** Open VSTS for your project, then navigate to "Load Test" page.
+
+![](media/1.png)
+
+
+**Step 2.** Click on "New" and select "URL based test".
+
+![](media/2.png)
+
+
+**Step 3.** Give this load test a name, rename URL name into something more meaningful and enter the MRP solution's URL (`{YourDnsName}/mrp`) into the URL field.
+
+![](media/3.png)
+
+
+**Step 4.** Click on "Add URL", give this URL a meaningful name and enter `{YourDnsName}/quotes?name` as a URL. This query will call your API, which then queries database.
+
+![](media/4.png)
+
+
+**Step 5.** Click on "Settings" and enter the following settings:
+* Run duration (minutes): `20`
+* Load pattern: `Constant`
+* Max v-users: `600`
+* Warmup duration (seconds): `60`
+* Browser mix: leave default value.
+* Load location: Closest location to the datacenter where MRP solution is hosted.
+
+![](media/5.png)
+
+
+**Step 6.** Click on "Save", then "Run test".
+>**Note:** The 2 and/or 3 task(s) has to be done during the load test, so it's a good idea to read through it before you click "Run Test" button.
+
+![](media/6.png)
+
+
+
+### Task 2: Inject faults into ACS environment using UI
+
+**Step 1.** Navigate to `http://localhost/marathon` to open up Marathon service.
+
+**Step 2.** Open "pumrp" folder and then "front".
+
+![](media/8.png)
+
+![](media/9.png)
+
+**Step 3.** Click on "web" application.
+
+![](media/10.png)
+
+**Step 4.** Select the application's instance and click on "Kill". At this stage Marathon will kill this instance and create one almost instantly. Every instance is a container, so it takes almost no time to spin it up.
+
+![](media/11.png)
+
+The new instance will be hosted on a different machine or on the same one with a different port.
+
+![](media/15.png)
+
+**Step 5.** Similarly, repeat previous steps for "db" application in "database" folder and "order" application in "api" folder. Make sure to spread out fault injections across the 20 minute period of the load test. This will allow the system to stabilize in-between the fault injections.
+
+
+
+### Task 3: Inject faults into ACS environment using API
+
```

Comment:
Please add a note stating that we are doing the same thing we did in the previous step with the API instead of the UI. I'd like people doing the labs to see that there are alternative ways of performing the same operation.